lib/SIL/SIL.cpp

```diff
@@ -115,3 +115,18 @@ swift::getLinkageForProtocolConformance(const NormalProtocolConformance *C,
       return (definition ? SILLinkage::Public : SILLinkage::PublicExternal);
   }
 }
+
+bool isDeclVisibleExternally(const ValueDecl *decl, const SILModule *module) {
```

Comment:
I think you to qualify the definition here, like `bool swift::isDeclVisibleExternally()`. CI failed to compile this PR with this error:

```
Undefined symbols for architecture x86_64:
  "swift::isDeclVisibleExternally(swift::ValueDecl const*, swift::SILModule const*)", referenced from:
      (anonymous namespace)::SILGlobalOpt::run() in libswiftSILOptimizer.a(GlobalOpt.cpp.o)
      (anonymous namespace)::LetPropertiesOptPass::run() in libswiftSILOptimizer.a(LetPropertiesOpts.cpp.o)
      (anonymous namespace)::LetPropertiesOpt::isConstantLetProperty(swift::VarDecl*) in libswiftSILOptimizer.a(LetPropertiesOpts.cpp.o)
      (anonymous namespace)::GlobalPropertyOpt::getFieldEntry(swift::VarDecl*) in libswiftSILOptimizer.a(GlobalPropertyOpt.cpp.o)
      (anonymous namespace)::DeadFunctionElimination::findAnchorsInTables() in libswiftSILOptimizer.a(DeadFunctionElimination.cpp.o)
      isDefaultCaseKnown(swift::ClassHierarchyAnalysis*, swift::FullApplySite, swift::ClassDecl*, llvm::SmallVector<swift::ClassDecl*, 8u>&) in libswiftSILOptimizer.a(SpeculativeDevirtualizer.cpp.o)
      isKnownFinalClass(swift::ClassDecl*, swift::SILModule&, swift::ClassHierarchyAnalysis*) in libswiftSILOptimizer.a(Devirtualize.cpp.o)
      ...
ld: symbol(s) not found for architecture x86_64
```

Reply:
I'm looking into this.

Reply:
Fixed and rebased.